test/tests/LayoutGrids.js

```diff
@@ -0,0 +1,593 @@
+'use strict';
+
+const { ariaTest } = require('..');
+const { By } = require('selenium-webdriver');
+
+const reload = async (session) => {
+  return session.get(await session.getCurrentUrl());
+};
+
+const clickUntilDisabled = async (session, selector) => {
+  const el = await session.findElement(By.css(selector));
+  await el.click();
+
+  if (await el.isEnabled()) {
+    return clickUntilDisabled(session, selector);
+  }
+};
+
+const checkActiveElement = function (/* gridcellsSelector, index */) {
+  let gridcellsSelector = arguments[0];
+  let index = arguments[1];
+  let gridcells = document.querySelectorAll(gridcellsSelector);
+  let gridcell = gridcells[index];
+  return (document.activeElement === gridcell) || gridcell.contains(document.activeElement);
+};
+
+const findFocusable = function (/* selector */) {
+  const selector = arguments[0];
+  const original = document.activeElement;
+  const focusable = Array.from(document.querySelectorAll(selector))
+    .map((parent) => {
+      const candidates = [parent, ...parent.querySelectorAll('*')];
+      for (let candidate of candidates) {
+
+        candidate.focus();
+        if (document.activeElement === candidate) {
+          return candidate;
+        }
+      }
+    }).filter((el) => !!el);;
+
+  original.focus();
+
+  return focusable;
+};
+
+const focusWithin = function (/* element */) {
+  // Assumption: `element is` focusable or contains only one focusable element.
+  let element = arguments[0];
+  let candidates = [element, ...element.querySelectorAll('*')];
+
+  for (let candidate of candidates) {
+    candidate.focus();
+
+    if (document.activeElement === candidate) {
+      return candidate;
+    }
+  }
+};
+
+let pageExamples = {
+  'ex1': {
+    gridSelector: '#ex1 [role="grid"]'
+  },
+  'ex2': {
+    gridSelector: '#ex2 [role="grid"]'
+  },
+  'ex3': {
+    gridSelector: '#ex3 [role="grid"]'
+  }
+};
+
+// Attributes
+
+ariaTest('Test "role=grid" attribute exists',
+  'grid/LayoutGrids.html', 'grid-role', async (t) => {
+
+    t.plan(3);
+
+    for (let exId in pageExamples) {
+      let ex = pageExamples[exId];
+      let gridLocator = t.context.By.css(ex.gridSelector);
+
+      t.is(
+        (await t.context.session.findElements(gridLocator)).length,
```

Comment:
It's not clear to me what you would like test for "role" attributes -- should I simply be testing whether an appropriate element type with an appropriate role exists, as I have in these tests for `role=grid`, `row` and `gridcell`?